extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/OpenTelemetryDestroyer.java

```diff
@@ -1,21 +1,36 @@
 package io.quarkus.opentelemetry;
 
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.time.Duration;
 import java.util.Map;
 
 import jakarta.enterprise.context.spi.CreationalContext;
 
+import org.eclipse.microprofile.config.ConfigProvider;
+
 import io.opentelemetry.api.OpenTelemetry;
 import io.opentelemetry.sdk.OpenTelemetrySdk;
 import io.quarkus.arc.BeanDestroyer;
+import io.smallrye.config.SmallRyeConfig;
 
 public class OpenTelemetryDestroyer implements BeanDestroyer<OpenTelemetry> {
     @Override
     public void destroy(OpenTelemetry openTelemetry, CreationalContext<OpenTelemetry> creationalContext,
             Map<String, Object> params) {
         if (openTelemetry instanceof OpenTelemetrySdk) {
+            // between flush and shutdown we will wait shutdown-wait-time, at the most.
+            var waitTime = getShutdownWaitTime().dividedBy(2);
             var openTelemetrySdk = ((OpenTelemetrySdk) openTelemetry);
-            openTelemetrySdk.getSdkTracerProvider().forceFlush();
-            openTelemetrySdk.getSdkTracerProvider().shutdown();
+            openTelemetrySdk.getSdkTracerProvider().forceFlush().join(waitTime.getSeconds(), SECONDS);
+            openTelemetrySdk.getSdkTracerProvider().shutdown().join(waitTime.getSeconds(), SECONDS);
```

Comment:
should we change this to:
```
.join(waitTime.toMillis(), MILLISECONDS);
```
if the user provides a value in milliseconds, or the default `1s` second divided by `2` could result in `0`

Reply:
sure, it's more flexible